pkg/certificates/ca_test.go

```diff
@@ -835,3 +998,263 @@ func TestLoadCertificatesNegative(t *testing.T) {
 		})
 	}
 }
+
+func TestLoadFromPEMFile(t *testing.T) {
+	type args struct {
+		filename string
+	}
+
+	errorSettingUpTypeFormatString := "Error setting up %s: %v"
+
+	certificateTestFilename := "certificate_test_filename"
+	goodCertificate, err := setupGoodCertificate()
+	if err != nil {
+		t.Errorf(errorSettingUpTypeFormatString, "certificate", err)
+	}
+
+	certificateRequestTestFilename := "certificate_request_test_filename"
+	goodCertificateRequest, err := setupGoodCertificateRequest()
+	if err != nil {
+		t.Errorf(errorSettingUpTypeFormatString, "certificate request", err)
+	}
+
+	rsaPrivateKeyTestFilename := "rsa_private_key_test_filename"
+	goodRSAPrivateKey, err := setupGoodRSAPrivateKey()
+	if err != nil {
+		t.Errorf(errorSettingUpTypeFormatString, "rsa private key", err)
+	}
+
+	privateKeyTestFilename := "private_key_test_filename"
+	goodPrivateKey, err := setupGoodPrivateKey()
+	if err != nil {
+		t.Errorf(errorSettingUpTypeFormatString, "private key", err)
+	}
+
+	publicKeyTestFilename := "public_key_test_filename"
+	goodPublicKey, err := setupGoodPublicKey()
+	if err != nil {
+		t.Errorf(errorSettingUpTypeFormatString, "public key", err)
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    []interface{}
+		wantErr bool
+	}{
+		{
+			name: "Certificate",
+			args: args{
+				filename: certificateTestFilename,
+			},
+			want: []interface{}{
+				goodCertificate,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Certificate Request",
+			args: args{
+				filename: certificateRequestTestFilename,
+			},
+			want: []interface{}{
+				goodCertificateRequest,
+			},
+			wantErr: false,
+		},
+		{
+			name: "RSA Private Key",
+			args: args{
+				filename: rsaPrivateKeyTestFilename,
+			},
+			want: []interface{}{
+				goodRSAPrivateKey,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Private Key",
+			args: args{
+				filename: privateKeyTestFilename,
+			},
+			want: []interface{}{
+				goodPrivateKey,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Public Key",
+			args: args{
+				filename: publicKeyTestFilename,
+			},
+			want: []interface{}{
+				goodPublicKey,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			o := mock_certificates.NewMockOser(ctrl)
+			switch tt.args.filename {
```

Comment:
This whole switch case can be swaped for a function pointer in the test case list, something like this: https://github.com/ansible/receptor/blob/devel/pkg/workceptor/workceptor_test.go#L140